src/selection_stream/selection_stream_processor.ts

```diff
@@ -338,13 +339,17 @@ export class SelectionStreamProcessor
 
     // The whole sheet is selected, select the anchor cell
     if (isEqual(this.anchor.zone, this.getters.getSheetZone(sheetId))) {
-      return this.selectZone({ ...anchor, zone: positionToZone(anchor.cell) });
+      return this.selectZone(
+        { ...anchor, zone: positionToZone(anchor.cell) },
+        "overrideSelection",
+        "AlterZoneCorner"
+      );
     }
 
     const tableZone = this.expandZoneToTable(anchor.zone);
 
     return !deepEquals(tableZone, anchor.zone)
-      ? this.selectZone({ ...anchor, zone: tableZone })
+      ? this.selectZone({ ...anchor, zone: tableZone }, "overrideSelection", "AlterZoneCorner")
```

Comment:
The only differences between the event type `AlterZoneCorner` and `ZonesSelected` is the handling in the `SheetView` plugin (`AlterZoneCorner` doesn't scroll). So I'm pretty sure that's what we want to use.

However now the name `AlterZoneCorner` is kinda bad, we change the whole selection zone here and not its corner. Maybe change it to just `AlterZone` ? Still its confusing that the difference between types `AlterZone` and `ZonesSelected` is how the `SheetView` plugin react to them...

Reply:
I agree with the renaming `AlterZoneCorner` => `AlterZone`